File: Data Structures & Algorithms/string-encode-and-decode/submission-5.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        result = ""
        for s in strs: 
            result += str(len(s))
            result += ';'
            result += s
        return result

    def decode(self, s: str) -> List[str]:
        length = ''
        result = []
        i = 0
        while i < len(s): 
            if s[i] == ';': 
                length = int(length)
                result.append(s[i + 1: i + 1 + length])
                i = i + 1 + length
                length = ''
            else: 
                length += s[i]
                i += 1
        return result

File: Data Structures & Algorithms/max-water-container/submission-5.ts
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights: number[]): number {
        let p1 = 0;
        let p2 = heights.length - 1;
        let result = 0;
        while(p1 < p2){
            let area = Math.min(heights[p1], heights[p2]) * (p2 - p1);
            result = Math.max(result, area);
            if (heights[p1] > heights[p2]){
                p2 -= 1
            }
            else{
                p1 += 1
            }
        }
        return result;
    }
}
